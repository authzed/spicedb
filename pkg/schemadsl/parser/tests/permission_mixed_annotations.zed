definition mydefinition {
    relation viewer: user
    
    permission view: user = viewer
    permission edit = viewer
    permission admin: group = viewer + edit
}