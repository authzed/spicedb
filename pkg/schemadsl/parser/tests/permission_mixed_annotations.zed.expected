NodeTypeFile
  end-rune = 165
  input-source = permission mixed annotations test
  start-rune = 0
  child-node =>
    NodeTypeDefinition
      definition-name = mydefinition
      end-rune = 165
      input-source = permission mixed annotations test
      start-rune = 0
      child-node =>
        NodeTypeRelation
          end-rune = 50
          input-source = permission mixed annotations test
          relation-name = viewer
          start-rune = 30
          allowed-types =>
            NodeTypeTypeReference
              end-rune = 50
              input-source = permission mixed annotations test
              start-rune = 47
              type-ref-type =>
                NodeTypeSpecificTypeReference
                  end-rune = 50
                  input-source = permission mixed annotations test
                  start-rune = 47
                  type-name = user
        NodeTypePermission
          end-rune = 90
          input-source = permission mixed annotations test
          relation-name = view
          start-rune = 61
          compute-expression =>
            NodeTypeIdentifier
              end-rune = 90
              identifier-value = viewer
              input-source = permission mixed annotations test
              start-rune = 85
          type-annotations =>
            NodeTypeTypeAnnotation
              end-rune = 81
              input-source = permission mixed annotations test
              start-rune = 78
              annotation-types =>
                NodeTypeIdentifier
                  end-rune = 81
                  identifier-value = user
                  input-source = permission mixed annotations test
                  start-rune = 83
        NodeTypePermission
          end-rune = 119
          input-source = permission mixed annotations test
          relation-name = edit
          start-rune = 96
          compute-expression =>
            NodeTypeIdentifier
              end-rune = 119
              identifier-value = viewer
              input-source = permission mixed annotations test
              start-rune = 114
        NodeTypePermission
          end-rune = 163
          input-source = permission mixed annotations test
          relation-name = admin
          start-rune = 125
          compute-expression =>
            NodeTypeUnionExpression
              end-rune = 163
              input-source = permission mixed annotations test
              start-rune = 151
              left-expr =>
                NodeTypeIdentifier
                  end-rune = 156
                  identifier-value = viewer
                  input-source = permission mixed annotations test
                  start-rune = 151
              right-expr =>
                NodeTypeIdentifier
                  end-rune = 163
                  identifier-value = edit
                  input-source = permission mixed annotations test
                  start-rune = 160
          type-annotations =>
            NodeTypeTypeAnnotation
              end-rune = 147
              input-source = permission mixed annotations test
              start-rune = 143
              annotation-types =>
                NodeTypeIdentifier
                  end-rune = 147
                  identifier-value = group
                  input-source = permission mixed annotations test
                  start-rune = 149